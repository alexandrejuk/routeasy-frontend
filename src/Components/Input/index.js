import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

import styles from './style.module.css'

const Input = ({
  error,
  disable,
  placeholder,
  name,
  onChange,
  type,
  value,
}) => (
  <input
    className={classNames(styles.input, { [styles.error]: error })}
    disabled={disable}
    name={name}
    onChange={onChange}
    placeholder={placeholder}
    type={type}
    value={value}
  />
)

Input.propTypes = {
  error: PropTypes.string,
  disable: PropTypes.bool,
  label: PropTypes.string,
  name: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  placeholder: PropTypes.string,
  type: PropTypes.oneOf([
    'number',
    'text',
  ]).isRequired,
  value: PropTypes.any.isRequired,
}

Input.defaultProps = {
  error: null,
  disable: false,
  label: null,
  placeholder: null,
}

export default Input
