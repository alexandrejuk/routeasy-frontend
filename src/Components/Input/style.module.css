@import '../../assets/css/colors.css';

.input {
  border: 1px solid var(--silver-chalice);
  border-radius: 3px;
  box-sizing: border-box;
  color: var(--silver-chalice);
  cursor: pointer;
  font-size: 0.9em;
  font-weight: 600;
  height: 35px;
  padding: 4px;
  width: 100%;
}

.input::placeholder {
  color: var(--silver-chalice);
}

.error {
  border-color: var(--pomegranate);
  color: var(--pomegranate);
}

.error::placeholder {
  color: var(--pomegranate);
}
