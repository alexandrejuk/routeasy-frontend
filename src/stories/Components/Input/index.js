import React from 'react'
import { Input } from '../../../Components'
import { action } from '@storybook/addon-actions/dist';

export default {
  title: 'Components/Input',
  component: Input,
}

const Template = ({
  disable,
  name,
  onChange,
  placeholder,
  type,
  value,
}) => (
  <Input
    disable={disable}
    name={name}
    onChange={onChange}
    placeholder={placeholder}
    type={type}
    value={value}
  />
)

export const inputNumber = Template.bind({})

inputNumber.args = {
  name: 'inputName',
  onChange: action('onChange!'),
  placeholder: 'Input Number',
  type: 'number',
  value: '',
}

export const inputText = Template.bind({})

inputText.args = {
  name: 'inputName',
  onChange: action('onChange!'),
  placeholder: 'Input Text',
  type: 'text',
  value: '',
}


export const inputNumberDisable = Template.bind({})

inputNumberDisable.args = {
  disable: true,
  name: 'inputName',
  onChange: action('onChange!'),
  placeholder: 'Input Number Disable',
  type: 'number',
  value: '',
}

export const inputTextDisable = Template.bind({})

inputTextDisable.args = {
  disable: true,
  name: 'inputName',
  onChange: action('onChange!'),
  placeholder: 'Input Text Disable',
  type: 'text',
  value: '',
}
