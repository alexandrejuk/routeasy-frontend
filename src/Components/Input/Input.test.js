import React from 'react'
import { render } from '@testing-library/react'
import userEvent from '@testing-library/user-event'

import Input from './'

describe('Input component', () => {
  it('Input should render and be updated correctly', () => {
    let currentValue = ''

    function callback (newValue) {
      currentValue = newValue
    }

    const { container, getByDisplayValue } = render(
      <Input
        name="InputName"
        onChange={callback}
        type="text"
        placeholder="InputName"
        value=""
      />
    )

    const input = getByDisplayValue(currentValue)

    expect(currentValue).toBe('')
    userEvent.type(input, 'Alexandre')
    expect(container).toMatchSnapshot()
  })
})
