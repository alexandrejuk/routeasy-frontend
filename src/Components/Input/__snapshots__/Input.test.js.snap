// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component Input should render and be updated correctly 1`] = `
<div>
  <input
    class="input"
    name="InputName"
    placeholder="InputName"
    type="text"
    value=""
  />
</div>
`;
